tests/rptest/tests/leadership_transfer_test.py

```diff
@@ -91,3 +92,56 @@ def get_partition():
                    err_msg="No partition with leader available")
 
         return partition[0]
+
+
+class AutomaticLeadershipBalancing(RedpandaTest):
+    topics = (TopicSpec(partition_count=100, replication_factor=3), )
+
+    def _get_leaders_by_node(self):
+        kc = KafkaCat(self.redpanda)
+        md = kc.metadata()
+        topic = next(filter(lambda t: t["topic"] == self.topic, md["topics"]))
+        leaders = (p["leader"] for p in topic["partitions"])
+        return collections.Counter(leaders)
+
+    @cluster(num_nodes=3)
+    def test_automatic_rebalance(self):
+        def all_partitions_present(num_nodes, min_per_node=None):
+            leaders = self._get_leaders_by_node()
+            count = sum(leaders[l] for l in leaders)
+            total = len(leaders) == num_nodes and count == 100
+            if min_per_node is not None:
+                min_sat = all((leaders[l] > min_per_node for l in leaders))
+                return total and min_sat
+            return total
+
+        # wait until all the partition leaders are elected
+        # on all three nodes
+        wait_until(lambda: all_partitions_present(3),
+                   timeout_sec=30,
+                   backoff_sec=2,
+                   err_msg="Leadership did not stablize")
+
+        # stop node and wait for all leaders to transfer
+        # to another node
+        node = self.redpanda.nodes[0]
+        self.redpanda.stop_node(node)
+        wait_until(lambda: all_partitions_present(2),
+                   timeout_sec=30,
+                   backoff_sec=2,
+                   err_msg="Leadership did not move to running nodes")
+
+        leaders = self._get_leaders_by_node()
+        assert self.redpanda.idx(node) not in leaders
+
+        # sleep for a bit to avoid triggering any
+        # of the sticky leaderhsip optimizations
+        time.sleep(60)
+
+        # restart the stopped node and wait for at least
+        # 10 leaders to be rebalanced on to the node
```

Comment:
I made it more strict. We'll have to keep an eye on it in CI. combined with partition rebalancing feature and that the error calculation in the leader balancer seems to not be very good when we have very few partitions it might be a little flaky. but its still better than it was!